Sources/MlemMiddleware/API Client/ApiClient+General.swift

```
@@ -12,6 +12,17 @@ public extension ApiClient {
         myInstance?.administrators.contains(where: { $0.id == myPerson?.id }) ?? false
     }
     
+    /// Returns true if both myPerson and the given person are admins on this instance and myPerson outranks the given person, false otherwise
+    func isHigherAdmin(than person: any Person1Providing) -> Bool {
+        guard person.api.actorId == actorId,
+              let myPerson,
+              let myAdminIndex = myInstance?.administrators.firstIndex(of: myPerson.person2),
+              let targetAdminIndex = myInstance?.administrators.firstIndex(where: { $0.actorId == person.actorId } ) else {
+            return false
+        }
+        return myAdminIndex < targetAdminIndex
```

This relies on the presence of `myInstance`, which isn't ideal. If `myInstance` isn't present, it'll fail silently and return `false`. 

I'm happy to leave this as-is for now as it isn't particularly important. One potential way we could fix this would be to add an `administratorIds: [Int]` property to `ApiClient.Context`, and `await` that property inside of `isHigherAdmin`. Although, unlike the already existing `ApiClient.Context` properties, `administratorIds` would be liable to change after the context is first fetched. So, we'd have to make sure to keep it updated whenever the administrator list changes. That solution means having more than one source of truth for the admin list, which isn't ideal, but I think it's better than what we have now.

---

Hmm, that's a tricky one. I _really_ don't want to make this function `async` because it's needed for the `ActionGroup`.

I think it's fine as-is because `myInstance` is _almost_ guaranteed to be defined--the `Session` initializers at the frontend are called at launch and account switch, they in turn call `getMyPerson`, and that sets `myInstance`, so unless that call fails (in which case we at least show an error toast) we can rely on `myInstance` being present.

---

Ah, I see. That's fine by me, then 👍